src/esp32/pulse_counter.rs

```diff
@@ -0,0 +1,50 @@
+use esp_idf_sys::{pcnt_isr_service_install, pcnt_isr_service_uninstall, EspError, ESP_OK};
+use std::sync::atomic::{AtomicBool, AtomicU32, Ordering};
+use std::sync::Arc;
+
+/*
+This module exists because we want to ensure uniqueness of unit number
+across instances of an Esp32 Pulse Counter unit and make sure the isr service
+is only installed once.
+
+THIS MODULE IS A TEMPORARY MEASURE. When abstracting the atomicity of Esp32
+peripherals to board, this logic should be moved there.
+
+TODO: v5 of ESP-IDF has refactored pulse counter to manage what this module
+accomplishes for us. Potentially only use this module when on chips on v4.
+
+*/
+
+lazy_static::lazy_static! {
+    static ref NEXT_UNIT: Arc<AtomicU32> = Arc::new(AtomicU32::new(0));
```

Comment:
Feels weird to wrap an atomic inside an Arc, would a zero size type make more sense?

Reply:
I'm don't quite see how to use a zero size type to produce the same functionality. However, I can explain the Arc. Atomics are thread-safe but don't allow shared ownership and that led to some problems whose nature I don't quite remember.

Reply:
My understanding is that the Arc count will be the same as the Atomic variable to you might be able to get away with just using the arc count and not the inner atomic

Reply:
None of the methods for returning the reference count on an Arc appear to be thread-safe, is this ok?